change/@ni-spright-blazor-21b29346-f22b-4578-81b8-7d76855eca6a.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
Probably in a follow-up PR to avoid growing scope, but how does nuget / assembly dependencies work? I expect SprightBlazor needs a dependency on NimbleBlazor for enums, etc at a minimum and potentially also utilities like [AttributeHelpers](https://github.com/ni/nimble/blob/c56b8518de39d66a1c86837b160348def0ad992d/packages/nimble-blazor/NimbleBlazor/AttributeHelpers.cs). Though for utilities we could go the copy route or add an additional assembly for shared functionality, but the enums question still stands.

Reply:
My assumption is that `SprightBlazor` won't need anything defined by `NimbleBlazor`. Presumably, anything it needs from Nimble will come from `nimble-components` (included in `spright-components`'s all-components bundle). Am I missing something? A client using Spright will likely need dependencies on both the `NimbleBlazor` and `SprightBlazor` and pull in both nugets.

If our `SprightBlazor` API eventually exposes some enum that is defined by `nimble-components`, it is probably simpler just to recreate the enum in the `SprightBlazor` namespace than to depend on the one from `NimbleBlazor`.

For utilities like AttributeHelpers, we can probably just copy them until there is critical mass for making a shared assembly. I don't see any others besides that one.

Reply:
BTW, I tried adding Spright Rectangle to the Blazor example app, and to my surprise, it worked without any trouble. I was bracing myself to see some redefinition errors from using both the Nimble and Spright all-component bundles, but that didn't seem to happen. Probably needs some more investigation to understand why, or at least to be confident that clients can (always) use `SprightBlazor` and `NimbleBlazor` together without issue.

Reply:
> BTW, I tried adding Spright Rectangle to the Blazor example app, and to my surprise, it worked without any trouble.

Sure... they do a [naive sanity check](https://github.com/microsoft/fast/blob/9c6dbb66615e6d229fc0ebf8065a67f109139f26/packages/web-components/fast-element/src/components/fast-definitions.ts#L200) when registering, doesn't mean you are in a good state when you are trying to register things multiple times. And I strongly suspect object references that need to be shared between components, like design token references, will be broken / competing for defining tokens.

> If our SprightBlazor API eventually exposes some enum that is defined by nimble-components, it is probably simpler just to recreate the enum in the SprightBlazor namespace than to depend on the one from NimbleBlazor.

I'd like the Blazor owners to weigh in on that as well. I'm skeptical of that statement being made so absolutely but haven't tried to think it through.

Reply:
Remind me why Spright's all-components bundle has to include Nimble's? Seems like keeping them separate would work fine and avoid potential conflicts. Otherwise, we're implicitly tying each Spright version to a specific version of Nimble, which might result in confusing behavior differences based on whether a client is consuming nimble via `nimble-components` imports or Spright's bundle.

Reply:
Got my answer over a chat. I had forgotten that these bundles include all dependencies, so even without including Nimble components, we would still end up with two copies of FAST, etc., which is no bueno.

Reply:
> > BTW, I tried adding Spright Rectangle to the Blazor example app, and to my surprise, it worked without any trouble.
> 
> Sure... they do a [naive sanity check](https://github.com/microsoft/fast/blob/9c6dbb66615e6d229fc0ebf8065a67f109139f26/packages/web-components/fast-element/src/components/fast-definitions.ts#L200) when registering, doesn't mean you are in a good state when you are trying to register things multiple times. And I strongly suspect object references that need to be shared between components, like design token references, will be broken / competing for defining tokens.
> 
> > If our SprightBlazor API eventually exposes some enum that is defined by nimble-components, it is probably simpler just to recreate the enum in the SprightBlazor namespace than to depend on the one from NimbleBlazor.
> 
> I'd like the Blazor owners to weigh in on that as well. I'm skeptical of that statement being made so absolutely but haven't tried to think it through.

Yeah, I'm not sure how I feel about this. I guess at worst, a client will have to namespace the "same" enum for its Spright vs Nimble usages:

```html
    <SprightWidget Appearance="SprightBlazor.ButtonAppearance.Ghost">
        <NimbleButton Appearance="NimbleBlazor.ButtonAppearance.Outline">
    </SprightWidget>
```
For the most part, I'm guessing Spright APIs will typically not need enums that are defined in nimble-components, so the above scenario shouldn't be common.

Reply:
> For the most part, I'm guessing Spright APIs will typically not need enums that are defined in nimble-components, so the above scenario shouldn't be common.

That was my gut feeling, too, but it seems fairly harmless to have SprightBlazor depend on NimbleBlazor (just in case). I found that a project dependency results in the Spright nuget having a dependency to the Nimble nuget version >= <SprightBlazor-version>. I guess this is pretty reasonable when the projects in your solution are at the same version, but doesn't make much sense for us. Instead, I put a package dependency on NimbleBlazor, since that supports specifying the version. It should get updated by Renovate along with the rest of the nuget dependencies.

Reply:
> It should get updated by Renovate along with the rest of the nuget dependencies.

If that is what we really on I expect that to not behave well. Renovate runs once a month and is intended for external dependencies, not packages inside the monorepo build. 

I'd still prefer managing that change on a separate PR.

Reply:
-I want it separate because of it acts weird it is easier to revert
- I want you to validate that it doesn't pull from Nuget in a local build but is resolving to the build in the workspace

Reply:
Even just syncing the branch and running npm install causes dirty changes because the lock file is out of sync:
![image](https://github.com/ni/nimble/assets/1588923/03381d10-c0f0-4d86-adbd-ce15d6728c7d)

Soo yea, please follow the initial suggestion and don't make the change in this PR.

Reply:
Removed the dependency. Created #2047 to track the tech debt.